src/key.js

```diff
@@ -404,8 +404,10 @@ h.extend(syn.key, {
 		// all current browsers have which property to normalize keyCode/charCode
 		if (result.keyCode) {
 			result.which = result.keyCode;
+			result.key = key;
 		} else {
 			result.which = result.charCode;
+			result.key = key;
```

Comment:
This can be done once instead of repeating the logic across both conditional branches.